.gitignore

```diff
@@ -1,12 +1,94 @@
 *~
+
```

Comment:
I don't love bloating the gitignore file because it may make it more difficult to understand the garbage generated as part of the build process, which we should have rules to remove.

Anyone else have opinions?

Reply:
I think regardless this should be a separate PR.

If you open a new PR with this change, let's discuss there.

Reply:
Reverted.

Reply:
Our `.gitignore` appeared big because I don't think we updated ours when you moved everything into the `hail` subdirectory.